packages/smart-vaults/src/signers/PasskeySigner.sol

```diff
@@ -0,0 +1,34 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.23;
+
+import { WebAuthn } from "@web-authn/WebAuthn.sol";
+
+/**
+ * A signer backed by a passkey.
+ */
+struct PasskeySigner {
+    uint256 x;
+    uint256 y;
+}
+
+function decodePasskeySigner(bytes memory signer) pure returns (PasskeySigner memory) {
+    return abi.decode(signer, (PasskeySigner));
```

Comment:
Do we need to wrap this like we did for AccountSigner?

Reply:
If you mean `AccountSigner.wrap`, that is because `AccountSigner` is a user defined value type while `PasskeySigner` is a struct so it doesn't work the same way.